extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -130,33 +130,38 @@ async function runRemoteQueriesApiRequest(credentials: Credentials, ref: string,
-      if (invalidRepos.length > 0) {
-        void logger.log(`Invalid repos: ${invalidRepos.join(', ')}`);
-      }
-      if (reposWithoutDbUploads.length > 0) {
-        void logger.log(`Repos without DB uploads: ${reposWithoutDbUploads.join(', ')}`);
-      }
+    await validateRepositories(error, credentials, ref, language, repositories, query);
+  }
+}
 
-      if (invalidRepos.length + reposWithoutDbUploads.length === repositories.length) {
-        // Every repo is invalid in some way
-        void showAndLogErrorMessage('Unable to run query on any of the specified repositories.');
-        return;
-      }
+export async function validateRepositories(error: any, credentials: Credentials, ref: string, language: string, repositories: string[], query: string) {
+  if (typeof error.message === 'string' && error.message.includes('Some repositories were invalid')) {
+    const invalidRepos = error?.response?.data?.invalid_repos || [];
+    const reposWithoutDbUploads = error?.response?.data?.repos_without_db_uploads || [];
+    void logger.log('Unable to run query on some of the specified repositories');
+    if (invalidRepos.length > 0) {
+      void logger.log(`Invalid repos: ${invalidRepos.join(', ')}`);
+    }
+    if (reposWithoutDbUploads.length > 0) {
+      void logger.log(`Repos without DB uploads: ${reposWithoutDbUploads.join(', ')}`);
+    }
 
-      const popupMessage = 'Unable to run query on some of the specified repositories. [See logs for more details](command:codeQL.showLogs).';
-      const rerunQuery = await showInformationMessageWithAction(popupMessage, 'Rerun on the valid repositories only');
-      if (rerunQuery) {
-        const validRepositories = repositories.filter(r => !invalidRepos.includes(r) && !reposWithoutDbUploads.includes(r));
-        void logger.log(`Rerunning query on set of valid repositories: ${JSON.stringify(validRepositories)}`);
-        await runRemoteQueriesApiRequest(credentials, ref, language, validRepositories, query);
-      }
+    if (invalidRepos.length + reposWithoutDbUploads.length === repositories.length) {
+      // Every repo is invalid in some way
+      void showAndLogErrorMessage('Unable to run query on any of the specified repositories.');
+      return;
+    }
 
-    } else {
-      void showAndLogErrorMessage(error);
+    const popupMessage = 'Unable to run query on some of the specified repositories. [See logs for more details](command:codeQL.showLogs).';
+    const rerunQuery = await showInformationMessageWithAction(popupMessage, 'Rerun on the valid repositories only');
+    if (rerunQuery) {
+      const validRepositories = repositories.filter(r => !invalidRepos.includes(r) && !reposWithoutDbUploads.includes(r));
+      void logger.log(`Rerunning query on set of valid repositories: ${JSON.stringify(validRepositories)}`);
+      await runRemoteQueriesApiRequest(credentials, ref, language, validRepositories, query);
```

Comment:
The function is named `validateRepositories`, but here it is re-running the request. Either this function's name should be changed to reflect what it is doing, or this call should be removed from the function.

Probably simpler to just rename. How about something like `rerunFailedRepositories`? Or `attemptRerun`?

Reply:
Good point 😄 `attemptRerun` is more descriptive 👍🏽